src/main/java/ee/tuleva/onboarding/mandate/generic/MandateDtoDeserializer.java

```diff
@@ -0,0 +1,39 @@
+package ee.tuleva.onboarding.mandate.generic;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import ee.tuleva.onboarding.epis.mandate.details.MandateDetails;
+import ee.tuleva.onboarding.epis.mandate.details.MandateDetailsDeserializer;
+import ee.tuleva.onboarding.mandate.MandateType;
+import java.io.IOException;
+import java.time.Instant;
+
+public class MandateDtoDeserializer extends JsonDeserializer<MandateDto<? extends MandateDetails>> {
+  @Override
+  public MandateDto<? extends MandateDetails> deserialize(JsonParser p, DeserializationContext ctxt)
```

Comment:
`p` is not a good parameter name - prefer to use full names like `jsonParser` or `parser`.
Same for `ctxt`

Reply:
This was the inherited name from interface it implements, but we can make it more descriptive